### PR TITLE
feat: Create RaceTypeLevel

### PR DESCRIPTION
**Pull request by DonLe27**
Need to get some designs from Maggie first before styling

### CODE REVIEW

**Review by BryanPan342 — CHANGES_REQUESTED** [review]
Can we make the `_redirects` page into the `netlify.toml` file.. OR add a `scripts/postbuild.js` to create the `_redirects` file

**Comment by DonLe27** [comment]
> Can we make the `_redirects` page into the `netlify.toml` file.. OR add a `scripts/postbuild.js` to create the `_redirects` file

Yup, added to .toml file

File: src/components/TypeRacerLevel.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
+// Identical to level1 except function name and export.
```
Comment (BryanPan342) [inline src/components/TypeRacerLevel.tsx]:
```suggestion
```

File: src/components/App.tsx [hunk 2]
```diff
@@ -23,17 +23,23 @@ export default function App(): JSX.Element {
 
   if (!isSignedIn) {
     return (
-      <div id='authWrapper'>
-        <div className='text' id='signIn'> Sign In </div>
-        <StyledFirebaseAuth uiConfig={_firebase.uiConfig()} firebaseAuth={_firebase.auth()}/>
+      <div id="authWrapper">
+        <div className="text" id="signIn">
+          {' '}
+          Sign In{' '}
```
Comment (BryanPan342) [inline src/components/App.tsx:44]: what does this do?

File: src/components/TypeRacerLevel.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  // After finishing the game, there is a 3 second delay before going to the next level.
+  if (redirect) {
+    return <Redirect to={props.nextLevel} />;
+  }
+  // User beat the game
+  else if (input === prompt && !timeOut) {
+    setTimeout(() => setRedirect(true), 3000);
+    return (
+      <div>
+        <p style={{ textAlign: 'center' }}>Type Racing</p>
```
Comment (BryanPan342) [inline src/components/TypeRacerLevel.tsx]: i see this `style={{ textAlign: 'center' }}` a lot.. thoughts about making this a css class?

@UCLA-Creative-Labs/tech would like other ppl's opinion on this too

File: src/components/TypeRacerLevel.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      <form>
+        <textarea
+          value={prompt}
+          onChange={(e) => setPrompt(e.target.value)}
+          className="textPrompt"
+        />
+      </form>
+    </div>
+  );
+}
```
Comment (BryanPan342) [inline src/components/TypeRacerLevel.tsx]:
```suggestion
function TextPrompt(props: PromptProps): JSX.Element {
  return (
    <div>
      <form>
        <textarea
          value={props.prompt}
          onChange={(e) => props.setPrompt(e.target.value)}
          className="textPrompt"
        />
      </form>
    </div>
  );
}
```
for consistency

File: src/components/TypeRacerLevel.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+  nextLevel: string;
+}
+
+interface RaceTrackProps {
+  setRestart?: React.Dispatch<React.SetStateAction<boolean>>;
+  setTimeOut?: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+// Renders the car and timer. When timer runs out, it prompts user to restart
+const RaceTrack = React.memo(({ setRestart, setTimeOut }: RaceTrackProps) => {
```
Comment (BryanPan342) [inline src/components/TypeRacerLevel.tsx]:
```suggestion
const RaceTrack = React.memo((props: RaceTrackProps) => {
```
for consistency

File: src/components/TypeRacerLevel.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+              return (
+                <span style={{ position: 'absolute', width: '100%' }}>
+                  <p className="restartPrompt">
+                    Too slow...
+                    <button
+                      className="restartButton"
+                      onClick={() => {
+                        setRestart(true);
+                      }}
+                    >
```
Comment (BryanPan342) [inline src/components/TypeRacerLevel.tsx]: nit: one line this?

File: src/components/TypeRacerLevel.tsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+        />
+      </form>
+    </div>
+  );
+}
+
+function TypeRacerLevel(props: LevelProps): JSX.Element {
+  const [redirect, setRedirect] = useState(false);
+  const [input, setInput] = useState('');
+  const p = prompts[Math.floor(Math.random() * prompts.length)]; // Triggered weird comma dangle error
```
Comment (BryanPan342) [inline src/components/TypeRacerLevel.tsx]: maybe make this a function? i see u use it later on line 105

**Review by BryanPan342 — DISMISSED** [review]
@DonLe27 dude this is freaking dope i love it ❤️

Had some comments on the code after reading it! Just a couple of nits here and there

**Review by arjunr00 — DISMISSED** [review]
Can you merge `main` into this branch? Then if you can move this new level component into `src/components/Levels/` and make sure `RaceTypeLevel` has all of its components wrapped using `<Level>` with appropriate props that'd be gr8. (See DESIGN_GUIDELINES.md for more info)

(Also, for the merge conflict with `App`, just go with whatever was in `main` already)

**Review by BryanPan342 — DISMISSED** [review]
LGTM

**Comment by BryanPan342** [comment]
@mergifyio

**Comment by BryanPan342** [comment]
@Mergifyio refresh

**Review by BryanPan342 — APPROVED** [review]
BIG THUMBS UP

**Pull request merged**